api/model/storage/postgres/timeseries.go

```diff
@@ -262,27 +290,81 @@ func (f *TimeSeriesField) fetchRepresentationTimeSeries(categoryBuckets []*api.B
+		err := rows.Scan(&timeArr)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "failed to parse timeseries set from postgres")
+		}
+		for _, v := range timeArr {
+			result[v] = math.NaN()
+		}
+
+	}
+
+	return &result, &timeArr, nil
+}
+
+func (s *Storage) parseTimeSet(rows pgx.Rows) (*map[float64]float64, *[]float64, error) {
+	result := map[float64]float64{}
+	if rows != nil {
+		defer rows.Close()
+	}
+	timeArr := []float64{}
+	for rows.Next() { // should be only 1 row
+		err := rows.Scan(&timeArr)
+		if err != nil {
+			return nil, nil, errors.Wrap(err, "failed to parse timeseries set from postgres")
+		}
+		for _, v := range timeArr {
+			result[v] = math.NaN()
+		}
+	}
+	return &result, &timeArr, nil
+}
+
 // FetchTimeseries fetches a timeseries.
-func (s *Storage) FetchTimeseries(dataset string, storageName string, timeseriesColName string, xColName string, yColName string, timeseriesURI string, filterParams *api.FilterParams, invert bool) (*api.TimeseriesData, error) {
+func (s *Storage) FetchTimeseries(dataset string, storageName string, timeseriesColName string, xColName string, yColName string, timeseriesURI []string, filterParams *api.FilterParams, invert bool) (*map[string]*api.TimeseriesData, error) {
 	// create the filter for the query.
 	wheres := make([]string, 0)
 	params := make([]interface{}, 0)
 
-	wheres = append(wheres, fmt.Sprintf("\"%s\" = $1", timeseriesColName))
-	params = append(params, timeseriesURI)
+	// build ANY ARRAY values
+	paramString := ""
+	if len(timeseriesURI) == 0 {
+		return nil, errors.New("No timeseriesURIs passed in")
+	}
+	for _, v := range timeseriesURI {
+		paramString += "'" + v + "',"
+	}
+	paramString = paramString[:len(paramString)-1] // remove end comma
+	wheres = append(wheres, fmt.Sprintf("\"%s\" = ANY(ARRAY[%s]::text[])", timeseriesColName, paramString))
 
 	wheres, params = s.buildFilteredQueryWhere(dataset, wheres, params, "", filterParams, invert)
 	where := fmt.Sprintf("WHERE %s", strings.Join(wheres, " AND "))
 
 	// Get count by category.
-	query := fmt.Sprintf("SELECT timeline.TimeStamps, COALESCE(filteredEvents.Counts, 'NaN') FROM "+
-		"(SELECT DISTINCT \"%s\" as TimeStamps FROM %s) timeline LEFT JOIN "+
-		"(SELECT \"%s\" as TimeStamps, \"%s\" as Counts FROM %s %s ) filteredEvents "+
-		"ON timeline.TimeStamps = filteredEvents.TimeStamps ORDER BY timeline.TimeStamps",
-		xColName, storageName, xColName, yColName, storageName, where)
+	query := fmt.Sprintf("SELECT ARRAY_AGG(filteredEvents.TimeStamps ORDER BY filteredEvents.TimeStamps), ARRAY_AGG(COALESCE(filteredEvents.Counts, 'NaN') ORDER BY filteredEvents.TimeStamps), filteredEvents.series_key FROM "+
+		"(SELECT \"%s\" as TimeStamps, \"%s\" as Counts, \"%s\" as series_key FROM %s %s ) filteredEvents "+
+		"GROUP BY filteredEvents.series_key",
+		xColName, yColName, timeseriesColName, storageName, where)
 
 	// execute the postgres query
-	res, err := s.client.Query(query, params...)
+	res, err := s.client.Query(query)
```

Comment:
I'm a bit worried that we no longer pass in params to the query. Does the where clause not use any parameters anymore? What about when some filters are applied?